chef/cookbooks/network/recipes/default.rb

```diff
@@ -499,23 +499,40 @@ def kill_nic(nic)
                     interfaces: ifs,
                     nic: nic
                   })
-        notifies :run, "bash[wicked-ifreload-#{nic.name}]", :immediately
+        notifies :create, "ruby_block[wicked-ifreload-required]", :immediately
       end
     else
       file "/etc/sysconfig/network/ifroute-#{nic.name}" do
         action :delete
       end
     end
-    bash "wicked-ifreload-#{nic.name}" do
-      action :nothing
-      code <<-EOF
-        wicked ifcheck --changed --quiet #{nic.name}
-        rc=$?
-        if [[ $rc != 0 ]]; then
-          wicked ifreload #{nic.name}
-        fi
-      EOF
+  end
+
+  # This, when notified by the above "ifcfg" templates, sets the "action"
+  # of the "wicked-ifreload-all" to ":run". The purpose of doing it this
+  # way (instead of notifying the "wicked-ifreload-all" resource directly),
+  # is to make sure that the ifrelaod is only run once after all ifcfg
+  # file have been update and independent of how many of them were changed.
+  ruby_block "wicked-ifreload-required" do
+    block do
+      r = resources(bash: "wicked-ifreload-all")
+      a = Array.new(r.action)
+      a << :run unless a.include?(:run)
+      a.delete(:nothing) if a.include?(:run)
+      r.action(a)
```

Comment:
Wow, that's some pretty amazing creativity here :-) I'm tempted of giving a +1 because you're teaching me some new magic here.

Asking out loud about other options first, though:
- is it maybe possible to have a global variable that the `ruby_block` would change and that would be used in the `code` of the `bash` resource?
- what about using a file as a marker to do the action?
- and what about simply running the `bash` resource all the time? Since we start with a `ifcheck`, it should not hurt that much, should it?